test/e2e/e2e_test.go

```diff
@@ -38,14 +43,8 @@ func TestE2EFlow(t *testing.T) {
 	revisions, err := getRevisions(awaitility)
 	require.NoError(t, err)
 
-	verifyResources(t, awaitility, johnsmithName, revisions,
-		wait.UntilMasterUserRecordHasConditions(provisioned()),
-		wait.UntilUserAccountHasSpec(*expUaSpec),
-		wait.UntilUserAccountHasConditions(provisioned()))
-	verifyResources(t, awaitility, extrajohnName, revisions,
-		wait.UntilMasterUserRecordHasConditions(provisioned()),
-		wait.UntilUserAccountHasSpec(*expExtraUaSpec),
-		wait.UntilUserAccountHasConditions(provisioned()))
+	verifyResourcesProvisioned(t, awaitility, johnsmithName, revisions, wait.UntilUserAccountHasSpec(*expUaSpec))
+	verifyResourcesProvisioned(t, awaitility, extrajohnName, revisions, wait.UntilUserAccountHasSpec(*expExtraUaSpec))
```

Comment:
just asking - couldn't we move `wait.UntilUserAccountHasSpec(*expUaSpec)` into the `verifyResourcesProvisioned` and do the check for all cases as well - the condition should be still fulfilled, shouldn't it?

Reply:
Yes we could. I ended up re organizing the tests a bit - https://github.com/codeready-toolchain/toolchain-e2e/pull/51/commits/fa602c224831eff6e45ae93f94aba7670a238af7